apache/trafficcontrol
# apache/trafficcontrol is a Go module

## What does this PR (Pull Request) do?

- [x] This PR switches us to the Go module system, which makes fixing #4835 or #4380 possible (although this PR does NOT fix either of those issues). 
    * #4380 would be fixed as soon as a vX.Y.Z tag is pushed for a release that includes this PR.
    * This PR preserves *all* of our existing module versions. In order for #4380 to be fixed, a future PR would need to update our dependencies to use v2+ libraries. See afb7bddb8b for all modifications made to our vendored packages in this PR. The only changes in Go files are trivial comments.
    * 4 new dependencies were added (0d536bd851). Note that these are not actually new dependencies, they are previously-unvendored dependencies that only [`/test/router/dnssec`](/apache/trafficcontrol/tree/f1a06fce50/test/router/dnssec) uses:
        - github.com/nxadm/tail
        - github.com/onsi/ginkgo
        - github.com/onsi/gomega
        - gopkg.in/tomb.v1
    * Using Go modules means only vendoring the packages necessary to satisfy dependencies, even if that means vendoring an incomplete repo (although it always includes licenses). Before this PR, there were 102 vendored dependencies. This PR cuts it down to 41:
        ```shell script
        ls -d vendor/{?[^i],github.com/}*/* | wc -l
        ```

## Which Traffic Control components are affected by this PR?


- CDN in a Box
- Documentation
- Grove
- Traffic Control Client (Go)
- Traffic Monitor
- Traffic Ops
- Traffic Ops ORT
- Traffic Stats
- CI tests

## What is the best way to verify this PR?

* Make sure Traffic Control still all builds and works
* Run the various tests affected by this PR:
    - grove/integration_test
    - infrastructure/cdn-in-a-box/traffic_ops_integration_test
    - test/router/dnssec
    - tools/golang
    - traffic_monitor/tests
    - traffic_ops/app/bin/tests
    - traffic_ops/testing/compare

## The following criteria are ALL met by this PR


- [x] This PR updates tests
- [x] This PR ensures that database migration sequence is correct
